src/main/java/org/jfrog/hudson/util/ExtractorUtils.java

```diff
@@ -609,14 +609,20 @@ private static boolean isMavenResolutionConfigured(ResolverContext resolverConte
                         resolverContext.getResolverOverrider() instanceof MavenResolver);
     }
 
+    /**
+     * The token that combines the project name and unique number to create unique workspace directory.
+     */
+    private static final String WORKSPACELIST = System.getProperty("hudson.slaves.WorkspaceList");
+
     /**
      * Create a temporary directory under a given workspace
      * @param launcher
      * @param ws
      * @throws Exception
      */
     public static FilePath createAndGetTempDir(hudson.Launcher launcher, FilePath ws) throws Exception {
-        final FilePath tempDirPath = new FilePath(ws.getParent(), ws.getName() + "@tmp");
+        final String WORKSPACELIST = System.getProperty("hudson.slaves.WorkspaceList");
```

Comment:
This is my suggestion:
```suggestion
     public static FilePath createAndGetTempDir(hudson.Launcher launcher, final FilePath ws) throws Exception {
        return launcher.getChannel().call(new MasterToSlaveCallable<FilePath, IOException>() {
            public FilePath call() {
                String workspaceList = System.getProperty("hudson.slaves.WorkspaceList"); // The token that combines the project name and unique number to create unique workspace directory.
                final FilePath tempDir = ws.sibling(ws.getName() + Objects.toString(workspaceList, "@") + "tmp").child("artifactory");
                File tempDirFile = new File(tempDir.getRemote());
                tempDirFile.mkdirs();
                tempDirFile.deleteOnExit();
                return tempDir;
            }
        });
    }
```
Please let me know what do you think.